bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/handler/HiveBridgeHandler.java

```diff
@@ -0,0 +1,394 @@
+            logger.warn("Unable to communicate with Hive API: {}", e.getMessage());
+            return false;
+        }
+
+        statusCode = response.getStatus();
+
+        if (statusCode != HttpStatus.OK_200) {
+            String statusLine = response.getStatus() + " " + response.getReason();
+            logger.warn("Error communicating with Hive API: {}", statusLine);
+            return false;
+        }
+
+        HiveLoginResponse o = gson.fromJson(response.getContentAsString(), HiveLoginResponse.class);
+        if (o.sessions != null) {
+            configuration.put(CONFIG_TOKEN, o.sessions.get(0).sessionId);
+            token = o.sessions.get(0).sessionId;
+            return true;
+        }
+        logger.warn("Hive API did not provide token: {}", response.getContentAsString());
+
+        return false;
+    }
+
+    public void checkForDevices() {
+        if (online) {
+            ContentResponse response;
+            try {
+                response = client.newRequest("https://api-prod.bgchprod.info:443/omnia/nodes").method(HttpMethod.GET)
+                        .header("Accept", "application/vnd.alertme.zoo-6.1+json")
+                        .header("Content-Type", "application/vnd.alertme.zoo-6.1+json")
+                        .header("X-Omnia-Client", "Openhab 2").header("X-Omnia-Access-Token", token)
+                        .timeout(DISCOVER_TIMEOUT_SECONDS, TimeUnit.SECONDS).send();
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                if (e.getMessage().contains("WWW-Authenticate header")) {
+                    getToken();
+                } else {
+                    logger.warn("Failed to get new token from Hive API: {}", e.getMessage());
+                }
+                return;
+            }
+            int statusCode = response.getStatus();
+
+            if (statusCode == HttpStatus.UNAUTHORIZED_401) {
+                // Token expired, get a new one and try again
+                getToken();
+                statusCode = response.getStatus();
+            }
+
+            if (statusCode != HttpStatus.OK_200) {
+                String statusLine = response.getStatus() + " " + response.getReason();
+                logger.warn("Error while reading from Hive API: {}", statusLine);
+                return;
+            }
+
+            HiveNodes o = gson.fromJson(response.getContentAsString(), HiveNodes.class);
+
+            if (o.nodes.size() > 0) {
+                // Loop through the nodes and add thermostats but only if they are showing a temperature as the api
+                // reports lots of thermostats for some reason
+                HiveNode thermostat = null;
+                HiveNode systemInfo = null;
+                for (HiveNode node : o.nodes) {
+                    if (node.attributes != null && node.attributes.nodeType != null
+                            && node.attributes.nodeType.reportedValue.equals(THERMOSTAT_NODE_TYPE)
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)